htlcswitch/switch.go

```diff
@@ -1717,6 +1738,28 @@ out:
 			totalSatSent += diffSatSent
 			totalSatRecv += diffSatRecv
 
+		// The ack ticker has fired so if we have any settle/fail entries
+		// for a forwarding package to ack, we will do so here in a batch
+		// db call.
+		case <-s.cfg.AckEventTicker.Ticks():
+			// If the current set is empty, pause the ticker.
+			if len(s.pendingSettleFails) == 0 {
+				s.cfg.AckEventTicker.Pause()
+				continue
+			}
+
+			// Batch ack the settle/fail entries.
+			if err := s.ackSettleFail(s.pendingSettleFails...); err != nil {
+				log.Errorf("Unable to ack batch of settle/fails: %v", err)
+			}
+
+			log.Debugf("Acked %d settle fails: %v", len(s.pendingSettleFails),
+				spew.Sdump(s.pendingSettleFails))
+
+			// Reset the pendingSettleFails buffer while keeping acquired
+			// memory.
+			s.pendingSettleFails = s.pendingSettleFails[:0]
```

Comment:
should we only clear this if the write was successful? don't think it's too big of a deal tho, it will eventually be retried on restart or reconnect

Reply:
Agree - good catch